.github/workflows/build-lint-test.yml

```diff
@@ -11,20 +11,12 @@ jobs:
       matrix:
         node-version: [18.x, 20.x, 22.x]
     steps:
-      - uses: actions/checkout@v4
-      - name: Install Corepack via Node
-        uses: actions/setup-node@v4
-        with:
-          node-version-file: '.nvmrc'
-      - name: Install Yarn
-        run: corepack enable
-      - name: Install Node.js ${{ matrix.node-version }} and restore Yarn cache
-        uses: actions/setup-node@v4
+      - name: Checkout and setup environment
+        uses: MetaMask/action-checkout-and-setup@v1
         with:
+          is-high-risk-environment: false
           node-version: ${{ matrix.node-version }}
-          cache: 'yarn'
-      - name: Install dependencies via Yarn
-        run: yarn --immutable
+          cache-node-modules: ${{ matrix.node-version == '22.x' }}
```

Comment:
Why do we only want to cache on a specific version?

Reply:
The cache key is the same, so it would show warnings that the cache could not be saved if we enable caching for all versions.

Reply:
So do we not have caching for the other versions or we use the same cache?

Reply:
We use the same cache. AFAIK `node_modules` is the same for different Node.js versions.